projects/workflows-creator/src/lib/builder/builder.component.ts

```diff
@@ -429,8 +473,55 @@ export class BuilderComponent<E> implements OnInit, OnChanges {
    * It builds a new diagram, emits the new diagram, and then tells Angular to update the view
    */
   async updateDiagram() {
+    const nodes = [
+      ...this.eventGroups[0].children,
+      ...this.actionGroups[0].children,
+      ...this.elseActionGroups[0].children,
+    ];
+    let isValid =
+      !!this.eventGroups[0].children.length &&
+      (!!this.actionGroups[0].children.length ||
+        !!this.elseActionGroups[0].children.length);
+    if (isValid) {
+      for (const node of nodes) {
+        switch (node.node.getIdentifier()) {
+          case EventTypes.OnChangeEvent:
```

Comment:
as mentioned, the builder component should not have any code specific to event types